### PR TITLE
UNSTABLE : UPDATED CODE TO BETTER REFLECT THE PROJECT

### PR DESCRIPTION
**Pull request by kiritocode1**
This contribution achieves few quality of life improvements : 
- removed unstable SSL code to check for the string value of a `bool`, JSON.parse() can do that 
- Error is a Falsey value so checking if it's an error or an empty string in `backend/controllers/subscription.ts`  is a better approach 
- overall standard approach to express, which have been there since the beginining . 

what this PR, did not address but is a known improvement is : 
[] SQL injection Bug  in runQuery()  function , no security  preset is put in place which could be dangerous
[] the api  design in `Router.ts` is honestly requiring a rework  , as in redo , the idea being 
-  instead of having `/createsubscriptionlist` , why not just a POSTrequest to `/subs/` 
- instead of a delete to `/deletesubscriptionlist/:name` why not just a DELETE to `/subs/`, it would make more sense to do so .

### CODE REVIEW

File: backend/utils/postgres.ts [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
+ * 
+ */
+export async function runQuery(queryText: string, values: any[] = []): Promise<any[] | Error> {
+  
+
   try {
     // Use a connection from the pool to execute the query
     const res = await pool.query(queryText, values);
+    if (!res.rows || res.rows.length === 0) {
+      return new Error(`Query returned no results: ${queryText}`);
```
Comment (sarvesh371) [inline backend/utils/postgres.ts:36]: Do not throw error as there will be case in our code where we are checking if subscription present already so we cannot raise exception please

**Review by kiritocode1 — COMMENTED** [review]
OK

**Comment by kiritocode1** [comment]
@sarvesh371  bhai , jldi merge krdo

File: backend/config/index.ts [hunk 2]
```diff
@@ -21,4 +21,16 @@ const envVariables = z.object({
...
+
+const config : z.infer<typeof envVariables> = { 
+  DB_HOST : process.env.DB_HOST?? "",
+  DB_USER : process.env.DB_USER??"",
+  DB_DATABASE : process.env.DB_DATABASE??"",
+  DB_PASSWORD : process.env.DB_PASSWORD??"",
+  DB_PORT : Number(process.env.DB_PORT)??0,
+  SERVER_PORT : Number(process.env.SERVER_PORT)??0,
+  SSL : JSON.parse(process.env.SSL??"false")
+}
```
Comment (Codimow) [inline backend/config/index.ts:33]:
```
const config: z.infer<typeof envVariables> = {
	DB_HOST: process.env.DB_HOST?.trim() || "",
	DB_USER: process.env.DB_USER?.trim() || "",
	DB_DATABASE: process.env.DB_DATABASE?.trim() || "",
	DB_PASSWORD: process.env.DB_PASSWORD?.trim() || "",
	DB_PORT: parseInt(process.env.DB_PORT || "0", 10),
	SERVER_PORT: parseInt(process.env.SERVER_PORT || "0", 10),
	SSL: process.env.SSL === "true"
};



This uses default value handling and parsing numbers
```

File: backend/controllers/subscription.ts [hunk 3]
```diff
@@ -13,9 +13,16 @@ export async function CreateSubscriptionList(req: Request, res: Response): Promi
...
+            res.status(500).json({
+                status: "error",
+                data: [],
+                message: "Internal Server Error , subscriptionList is not a valid object",
+            });
+            return;
+        }
 
-        if ( !subscriptionList) {
+        if (subscriptionList.length ===0 ) {
```
Comment (Codimow) [inline backend/controllers/subscription.ts:25]: i think using subscriptionList.length > 0 is better to check the same names. Correct me if im wrong

Comment (kiritocode1) [reply backend/controllers/subscription.ts:25]: No it wouldn't be , a person could have 0 subscription , so >0 would only run for all others

File: backend/config/index.ts (hunk 2 above)
Comment (kiritocode1) [reply backend/config/index.ts:33]: nice to know , im going to make some changes to this code ,

Comment (kiritocode1) [reply backend/config/index.ts:33]: parseInt is not to be used , in general

File: backend/utils/postgres.ts [hunk 4]
```diff
@@ -1,29 +1,60 @@
...
+ */
+export async function runQuery(queryText: string, values: any[] = [], allowEmptyArray : boolean = false): Promise<any[] | Error> {
+  
+
   try {
     // Use a connection from the pool to execute the query
-    const res = await pool.query(queryText, values);
+    const res = await pool.query(queryText, values); 
+    if ((!res.rows || res.rows.length === 0) && !allowEmptyArray) {
+      return new Error(`Query returned no results: ${queryText}`);
```
Comment (sarvesh371) [inline backend/utils/postgres.ts:36]: Why are you throwing error here

having 0 rows in output does not mean it is a error. Please remove this

File: backend/utils/postgres.ts [hunk 5]
```diff
@@ -1,29 +1,60 @@
...
   }
 }
 
+
+// ? THIS IS COMMENTED OUT BECAUSE IT IS NOT NEEDED
 // Clean up pool (if needed, e.g., on app shutdown)
-export async function closePool(): Promise<void> {
-  await pool.end();
-}
+// export async function closePool(): Promise<void> {
```
Comment (sarvesh371) [inline backend/utils/postgres.ts:48]: Remove the commented code if not used

File: backend/controllers/subscription.ts [hunk 6]
```diff
@@ -11,8 +11,18 @@ export async function CreateSubscriptionList(req: Request, res: Response): Promi
             [name]
         );
 
-        // Return an error if the subscription already exists
-        if (subscriptionList && subscriptionList.length > 0) {
+        // Return an error if the subscription already exists. 
+        
+        if (subscriptionList instanceof Error) {
```
Comment (sarvesh371) [inline backend/controllers/subscription.ts:16]: There is no case where subscriptionList will be error

if (subscriptionList && subscriptionList.length > 0) {

**Pull request merged**